precompile/contracts/feemanager/simulated_test.go

```diff
@@ -0,0 +1,319 @@
+				allowlisttest.SetAsEnabled(t, backend, feeManager, admin, testContractAddr)
+
+				got, err := testContract.GetFeeConfig(nil)
+				require.NoError(t, err)
+				verifyFeeConfigsMatch(t, genesisFeeConfig, toCommonFeeConfig(got))
+
+				blockNum := setFeeConfig(t, backend, testContract, admin, cchainFeeConfig)
+
+				got, err = testContract.GetFeeConfig(nil)
+				require.NoError(t, err)
+				verifyFeeConfigsMatch(t, cchainFeeConfig, toCommonFeeConfig(got))
+
+				lastChangedAt, err := testContract.GetFeeConfigLastChangedAt(nil)
+				require.NoError(t, err)
+				require.Equal(t, blockNum, lastChangedAt.Uint64())
+			},
+		},
+		{
+			name: "should reject a transaction below the minimum fee",
+			test: func(t *testing.T, backend *sim.Backend, feeManager *feemanagerbindings.IFeeManager) {
+				testContractAddr, testContract := deployFeeManagerTest(t, backend, admin)
+				allowlisttest.SetAsEnabled(t, backend, feeManager, admin, testContractAddr)
+
+				// Get current min base fee
+				config, err := testContract.GetFeeConfig(nil)
+				require.NoError(t, err)
+				originalMinBaseFee := config.MinBaseFee
+
+				// Raise min fee by one
+				raisedConfig := genesisFeeConfig
+				raisedConfig.MinBaseFee = new(big.Int).Add(originalMinBaseFee, big.NewInt(1))
+				_ = setFeeConfig(t, backend, testContract, admin, raisedConfig)
+
+				// Try to send a transaction with the old (now too low) fee - should fail
+				lowFeeAuth := testutils.NewAuth(t, adminKey, params.TestChainConfig.ChainID)
+				lowFeeAuth.GasLimit = 100000
+				lowFeeAuth.GasFeeCap = originalMinBaseFee
+				lowFeeAuth.GasTipCap = big.NewInt(1)
+
+				_, err = testContract.SetFeeConfig(
+					lowFeeAuth,
+					genesisFeeConfig.GasLimit,
+					big.NewInt(int64(genesisFeeConfig.TargetBlockRate)),
+					originalMinBaseFee,
+					genesisFeeConfig.TargetGas,
+					genesisFeeConfig.BaseFeeChangeDenominator,
+					genesisFeeConfig.MinBlockGasCost,
+					genesisFeeConfig.MaxBlockGasCost,
+					genesisFeeConfig.BlockGasCostStep,
+				)
+				require.Error(t, err)
+				require.Contains(t, err.Error(), "underpriced")
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			backend := newBackendWithFeeManager(t)
+			defer backend.Close()
+
+			feeManager, err := feemanagerbindings.NewIFeeManager(feemanager.ContractAddress, backend.Client())
+			require.NoError(t, err)
+
+			tc.test(t, backend, feeManager)
+		})
+	}
+}
+
+func TestIFeeManager_Events(t *testing.T) {
+	chainID := params.TestChainConfig.ChainID
+	admin := testutils.NewAuth(t, adminKey, chainID)
+
+	backend := newBackendWithFeeManager(t)
+	defer backend.Close()
+
+	feeManager, err := feemanagerbindings.NewIFeeManager(feemanager.ContractAddress, backend.Client())
+	require.NoError(t, err)
+
+	t.Run("should emit FeeConfigChanged event", func(t *testing.T) {
+		require := require.New(t)
+
+		// Change fee config from genesis to C-Chain
+		tx, err := feeManager.SetFeeConfig(
```

Comment:
why we don't use the `setFeeConfig` helper here?

Reply:
What do you mean can we use the helper here? I am using the `SetFeeConfig` helper.

Reply:
Fixed.